src/main/java/uk/gov/laa/ccms/caab/controller/application/summary/EditGeneralDetailsSectionController.java

```
@@ -364,26 +394,209 @@ private void populateLinkedCasesConfirmDropdowns(final Model model) {
+   */
+  @GetMapping("/application/summary/linked-cases/search/results")
+  public String linkedCasesSearchResults(
+      @RequestParam(value = "page", defaultValue = "0") final int page,
+      @RequestParam(value = "size", defaultValue = "10") final int size,
+      @ModelAttribute(CASE_SEARCH_RESULTS) final List<BaseApplication> caseSearchResults,
+      final HttpServletRequest request,
+      final Model model,
+      final HttpSession session) {
+
+    // Paginate the results list, and convert to the Page wrapper object for display
+    final ApplicationDetails applicationDetails = applicationMapper.toApplicationDetails(
+        PaginationUtil.paginateList(Pageable.ofSize(size).withPage(page), caseSearchResults));
+
+    model.addAttribute(CURRENT_URL,  request.getRequestURL().toString());
+
+    model.addAttribute(CASE_RESULTS_PAGE, applicationDetails);
+    session.setAttribute(CASE_RESULTS_PAGE, applicationDetails);
+    return "application/summary/application-linked-case-search-results";
+  }
+
+  /**
+   * Handles the GET request to initiate the addition of a linked case based on
+   * selected case reference ID.
+   *
+   * @param caseReferenceId    The case reference ID path variable.
+   * @param applicationDetails Application details from session attribute.
+   * @param model              Spring MVC model.
+   * @return The view name for adding a linked case.
+   */
+  @GetMapping("/application/summary/linked-cases/{case-reference-id}/add")
+  public String addLinkedCaseGet(
+      @PathVariable("case-reference-id") final String caseReferenceId,
+      @SessionAttribute(CASE_RESULTS_PAGE) final ApplicationDetails applicationDetails,
+      final Model model
+  ) {
+
+    final BaseApplication baseApplication = applicationDetails.getContent() == null ? null :
+        applicationDetails.getContent().stream()
+            .filter(lc -> caseReferenceId.equals(lc.getCaseReferenceNumber()))
+            .findFirst()
+            .orElse(null);
```

Better to throw an exception if not found. The url could have been manipulated.

---

Amended